### PR TITLE
airflow audit logging

### PR DESCRIPTION
**Pull request by alexyu0**

### CODE REVIEW

**Comment by avenkatraman** [comment]
Can you please squash your commits before you merge?

File: airflow/models/log.py
Comment (avenkatraman) [inline airflow/models/log.py]: This comment reads a bit weird. What is it saying? And can you remind me when owner is set vs not set? I remember we looked at this together a while back.

File: airflow/version.py
Comment (avenkatraman) [inline airflow/version.py]: This should be bumped to `affirm10`

File: airflow/www_rbac/api/experimental/endpoints.py
Comment (avenkatraman) [inline airflow/www_rbac/api/experimental/endpoints.py:114]: What does this function do when `execution_date` is None? Are we meaningfully changing the behavior here?

File: airflow/www_rbac/api/experimental/endpoints.py
Comment (avenkatraman) [inline airflow/www_rbac/api/experimental/endpoints.py:313]: Is the owner the username of someone from the User table?

File: airflow/www_rbac/api/experimental/endpoints.py
Comment (avenkatraman) [inline airflow/www_rbac/api/experimental/endpoints.py:359]: This won't have PII in it right? Only if the payload has PII in it?

File: airflow/www_rbac/utils.py
Comment (avenkatraman) [inline airflow/www_rbac/utils.py:501]: Will this commit on exit of session?

File: airflow/www_rbac/decorators.py
Comment (avenkatraman) [inline airflow/www_rbac/decorators.py:37]: What is g? I'm curious if we can use this on the api requests as well rather than you building the logic there directly

Comment (avenkatraman) [reply airflow/www_rbac/decorators.py:37]: My assumption is that it's session based so it won't work?

File: airflow/www_rbac/views.py
Comment (avenkatraman) [inline airflow/www_rbac/views.py]: minor nit - can we move the event_name to the last kwarg before session so that the "core" args/kwargs come first? imo makes it more readable.

File: airflow/www_rbac/views.py
Comment (avenkatraman) [inline airflow/www_rbac/views.py:1006]: Is it not possible to just use decorators here? like for the action logging?

File: airflow/www_rbac/views.py
Comment (avenkatraman) [inline airflow/www_rbac/views.py:1146]: Why nix this? This is for a specific dagrun? Only one execution date of interest.

File: airflow/www_rbac/views.py
Comment (avenkatraman) [inline airflow/www_rbac/views.py:2576]: can you please DRY this up? only difference afaict is the first arg to Log.

File: airflow/www_rbac/views.py
Comment (avenkatraman) [inline airflow/www_rbac/views.py:2589]: reason for adding updating redirect?

File: airflow/www_rbac/views.py
Comment (avenkatraman) [inline airflow/www_rbac/views.py:2626]: same q

File: airflow/www_rbac/views.py
Comment (avenkatraman) [inline airflow/www_rbac/views.py:2694]: same q for all of these. just answer one i guess.. what happens without it that you didnt like

File: airflow/models/log.py
Comment (alexyu0) [reply airflow/models/log.py]: will clarify this. tldr is that the task is only set in the init method of the TaskInstance class so when the task_instance passed in is retrieved via sqlalchemy query, the task property isnt set and so the owner isn't discoverable that way. also we dont necessarily want the dag owner, but the person who issued the request

File: airflow/www_rbac/api/experimental/endpoints.py
Comment (alexyu0) [reply airflow/www_rbac/api/experimental/endpoints.py:114]: nope it just gets set later down the code path. setting it here means we can use it in the log entry

https://github.com/Affirm/airflow/blob/1.10.9-affirm/airflow/api/common/experimental/trigger_dag.py#L56

File: airflow/www_rbac/api/experimental/endpoints.py
Comment (alexyu0) [reply airflow/www_rbac/api/experimental/endpoints.py:359]: this is the payload so actually yes it could have PII if someone is calling this endpoint with an arg that has PII

File: airflow/www_rbac/decorators.py
Comment (alexyu0) [reply airflow/www_rbac/decorators.py:37]: yeah i tried using `g` initially and it didn't work. traced it back and realized that the API auth isn't tied into flask, its just a decorator that queries the metadata tables

File: airflow/www_rbac/utils.py
Comment (alexyu0) [reply airflow/www_rbac/utils.py:501]: yep 

https://github.com/Affirm/airflow/blob/1.10.9-affirm/airflow/utils/db.py#L50

File: airflow/www_rbac/api/experimental/endpoints.py
Comment (alexyu0) [reply airflow/www_rbac/api/experimental/endpoints.py:313]: yes

File: airflow/www_rbac/views.py
Comment (alexyu0) [reply airflow/www_rbac/views.py:1006]: i needed the `execution_date` and i don't think there's a way to provide a value to the decorator and then use that in the function body as well

File: airflow/www_rbac/views.py
Comment (alexyu0) [reply airflow/www_rbac/views.py:1146]: so idk why but airflow calls each of these cancel/clear functions twice when canceling/clearing. first one is a dry run and second is the actual one. that leads to it being logged twice each time we do the action. by removing it and just having it in the helper, it only gets logged once

File: airflow/www_rbac/views.py
Comment (alexyu0) [reply airflow/www_rbac/views.py:2694]: it would dump you back to the list page with none of your filters which was annoying. this way it keeps the filters you had set

File: airflow/www_rbac/api/experimental/endpoints.py
Comment (avenkatraman) [reply airflow/www_rbac/api/experimental/endpoints.py:114]: sg!

File: airflow/www_rbac/views.py
Comment (avenkatraman) [reply airflow/www_rbac/views.py:1006]: Not sure I follow. The only change we need to make is adding the Log DB record and the action_logging method that you edited above gets the execution_date just fine.

File: airflow/www_rbac/views.py
Comment (avenkatraman) [inline airflow/www_rbac/views.py]: This is actually so dumb. I dislike the Log implementation -- Like it's weird that they read the execution date from the task_instance but then also look at the kwargs. This code now looks more ugly even though it's doing basically the same thing.

File: airflow/www_rbac/views.py
Comment (avenkatraman) [inline airflow/www_rbac/views.py]: is `event_name` ever None? Afaict it is not.

File: airflow/www_rbac/views.py
Comment (avenkatraman) [reply airflow/www_rbac/views.py:1146]: I see - so the unconfirmed time it will log and then log the same thing again.

**Review by avenkatraman — APPROVED** [review]
I'm accepting so as to not block you after adjusting some of these. I think some stuff still looks a bit weird and can be improved. Happy to hop on  a zoom call to discuss. This won't really be blocked by code freeze anyway since it's to a resource we dont deploy often anyway..

File: airflow/www_rbac/views.py
Comment (alexyu0) [reply airflow/www_rbac/views.py]: its not but since its a kwarg, i feel like its nice to have the check so we dont log something with none as the event name

File: airflow/www_rbac/views.py
Comment (avenkatraman) [inline airflow/www_rbac/views.py]: Let's make this an arg not a kwarg

File: airflow/www_rbac/views.py
Comment (avenkatraman) [inline airflow/www_rbac/views.py]: just pass execution_date here and kill the `if/else`

File: airflow/www_rbac/views.py
Comment (avenkatraman) [reply airflow/www_rbac/views.py]: if default doesnt make sense let's make it an kwarg.

**Pull request merged**
